source/server/http/admin.cc

```diff
@@ -163,7 +163,11 @@ Http::Code AdminImpl::handlerCpuProfiler(const std::string& url, Buffer::Instanc
 
   bool enable = query_params.begin()->second == "y";
   if (enable && !Profiler::Cpu::profilerEnabled()) {
-    Profiler::Cpu::startProfiler("/var/log/envoy/envoy.prof");
+    if (!Profiler::Cpu::startProfiler(profiler_path_)) {
+      response.add("?enable=<y|n>\n");
+      return Http::Code::BadRequest;
```

Comment:
This returns a 400. I think it's not actually a bad client formed request, it's an internal server error, so you want to return a 500.